gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientReadChannel.java

```diff
@@ -210,14 +211,17 @@ private class ContentReadChannel {
     // in-place seeks.
     private byte[] skipBuffer = null;
     private ReadableByteChannel byteChannel = null;
-    private boolean randomAccess;
+    private AdaptiveFileAccessPattern fileAccessPattern;
 
     public ContentReadChannel(
         GoogleCloudStorageReadOptions readOptions, StorageResourceId resourceId) {
       this.blobId =
           BlobId.of(
               resourceId.getBucketName(), resourceId.getObjectName(), resourceId.getGenerationId());
-      this.randomAccess = readOptions.getFadvise() == Fadvise.RANDOM;
+      this.fileAccessPattern = new AdaptiveFileAccessPattern(resourceId, readOptions);
+      if (gzipEncoded) {
+        fileAccessPattern.overrideAccessPattern(false);
```

Comment:
Why not have this as part of constructor. That way we can make the overRideAccessPattern final. Even better have a dummy noop FileAccessPattern for this case.

Reply:
Have a set method breaks the encapsulation. I think in this case it can be easily avoided.